Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift

```diff
@@ -188,8 +191,16 @@ open class ChatMessageListView: UITableView, Customizable, ComponentsProvider {
             switch $0 {
             case let .insert(message, index: index):
                 UIView.performWithoutAnimation {
-                    self.reloadData()
+                    self.performBatchUpdates {
+                        self.insertRows(at: [index], with: .none)
+                    } completion: { _ in
+                        // Update previous row to remove timestamp if needed
+                        // +1 instead of -1 because the message list is inverted
+                        let previousIndex = IndexPath(row: index.row + 1, section: index.section)
+                        self.reloadRows(at: [previousIndex], with: .none)
```

Comment:
Is OK for the case when the inserted message is the only message in the channel ie. when the previous cell we're reloading does not exist?

Reply:
Nice catch, again 👏